@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -999,17 +1002,69 @@ export class TextElementsRenderer {
 
         const groupedPriorityLists: Map<number, TextElementLists> = new Map();
 
+        // Cache for point labels which may have duplicates in same tile or in neighboring tiles.
+        const pointLabelCache: Map<string, TextElement> = new Map();
```

Comment:
Why not store the index into the uniqueLabels array instead. This will remove the need for the "indexOf" call below and you can still easily access the textElement.